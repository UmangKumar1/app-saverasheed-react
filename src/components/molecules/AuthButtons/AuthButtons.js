import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'react-emotion'

import muiTheme from '../../../theme'
import { mapObjAsArray } from '../../../util/ramda-extra'
import SignInButton from '../../atoms/SignInButton/SignInButton'
import Link from '../../atoms/Link'
import authProviders from './../../../fixtures/authProviders'

const { shape, string, func } = PropTypes

const {
  palette: {
    tertiary: { main: tertiaryMain, contrastText },
  },
} = muiTheme

const AuthButtonsRoot = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  color: tertiaryMain,
  marginBottom: 50,
})

const AuthButtonsTitle = styled('h1')({
  color: contrastText,
  textAlign: 'center',
  marginBottom: 30,
})

const authButtonCx = css({
  marginBottom: 10,
})

const linkCx = css({
  width: '100%',
  display: 'flex',
  flexDirection: 'column',
})

const AuthButtons = ({
  match: {
    url,
    params: { authType },
  },
  signIn,
}) => (
  <AuthButtonsRoot>
    <AuthButtonsTitle>
      {authType === 'signin' ? 'Sign In' : 'Sign Up'} with:
    </AuthButtonsTitle>
    {mapObjAsArray(
      auth =>
        auth.id === 'password' ? (
          <Link to={`${url}/email`} className={linkCx} key={auth.id}>
            <SignInButton auth={auth} className={authButtonCx} />
          </Link>
        ) : (
          <SignInButton
            key={auth.id}
            auth={auth}
            className={authButtonCx}
            onClick={() => signIn(auth.id, authType)}
          />
        ),
      authProviders,
    )}
  </AuthButtonsRoot>
)

AuthButtons.propTypes = {
  match: shape({
    url: string,
    params: shape({ authType: string }),
  }).isRequired,
  signIn: func,
}

export default AuthButtons
