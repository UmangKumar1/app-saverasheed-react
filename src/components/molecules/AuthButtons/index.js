export { default, default as AuthButtons } from './AuthButtons'
