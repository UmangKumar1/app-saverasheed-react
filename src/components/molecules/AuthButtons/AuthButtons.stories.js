import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import AuthButtons from '.'

storiesOf('AuthButtons', module)
  .add('sign in', () => (
    <AuthButtons
      match={{ url: '/main/login', params: { authType: 'login' } }}
      signIn={action('sign in!')}
    />
  ))
  .add('sign up', () => (
    <AuthButtons
      match={{ url: '/main/signup', params: { authType: 'signup' } }}
      signIn={action('sign in!')}
    />
  ))
