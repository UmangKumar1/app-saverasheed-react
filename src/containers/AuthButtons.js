import { connect } from 'react-redux'
import { compose } from 'recompose'

import AuthButtons from '../components/molecules/AuthButtons'
import { authenticateRequest } from '../modules/auth/actions'
import { subscribe } from '../util/reselect'

const mapStateToProps = subscribe({})

const actions = { signIn: authenticateRequest }

export default compose(
  connect(
    mapStateToProps,
    actions,
  ),
)(AuthButtons)
