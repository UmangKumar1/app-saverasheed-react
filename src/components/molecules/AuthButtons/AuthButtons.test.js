import React from 'react'
import { shallow } from 'enzyme'

import AuthButtons from '.'

const wrap = (props = {}) => shallow(<AuthButtons {...props} />)

describe('AuthButtons', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      match: { url: '/main/login', params: { authType: 'login' } },
      signIn: jest.fn(),
    })
    expect(wrapper).toMatchSnapshot()
  })
})
