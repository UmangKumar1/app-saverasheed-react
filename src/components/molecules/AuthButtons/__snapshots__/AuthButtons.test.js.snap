// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButtons renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthButtons
    match={
      Object {
        "params": Object {
          "authType": "login",
        },
        "url": "/main/login",
      }
    }
    signIn={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(h1)>
          Sign Up
           with:
        </Styled(h1)>,
        Array [
          <SignInButton
            auth={
              Object {
                "Icon": [Function],
                "id": "google.com",
                "method": "signInWithGoogle",
                "name": "Google",
              }
            }
            className="css-1r2f04i"
            onClick={[Function]}
          />,
          <SignInButton
            auth={
              Object {
                "Icon": [Function],
                "id": "facebook.com",
                "method": "signInWithFacebook",
                "name": "Facebook",
              }
            }
            className="css-1r2f04i"
            onClick={[Function]}
          />,
          <Styled(Link)
            className="css-4rmlxi"
            replace={false}
            to="/main/login/email"
          >
            <SignInButton
              auth={
                Object {
                  "Icon": [Function],
                  "id": "password",
                  "method": "signIn",
                  "name": "Email and password",
                }
              }
              className="css-1r2f04i"
            />
          </Styled(Link)>,
          <SignInButton
            auth={
              Object {
                "Icon": [Function],
                "id": "portis",
                "method": "",
                "name": "Portis wallet",
              }
            }
            className="css-1r2f04i"
            onClick={[Function]}
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Sign Up",
            " with:",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Sign Up",
          " with:",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "google.com",
        "nodeType": "function",
        "props": Object {
          "auth": Object {
            "Icon": [Function],
            "id": "google.com",
            "method": "signInWithGoogle",
            "name": "Google",
          },
          "className": "css-1r2f04i",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "facebook.com",
        "nodeType": "function",
        "props": Object {
          "auth": Object {
            "Icon": [Function],
            "id": "facebook.com",
            "method": "signInWithFacebook",
            "name": "Facebook",
          },
          "className": "css-1r2f04i",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "password",
        "nodeType": "class",
        "props": Object {
          "children": <SignInButton
            auth={
              Object {
                "Icon": [Function],
                "id": "password",
                "method": "signIn",
                "name": "Email and password",
              }
            }
            className="css-1r2f04i"
          />,
          "className": "css-4rmlxi",
          "replace": false,
          "to": "/main/login/email",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "auth": Object {
              "Icon": [Function],
              "id": "password",
              "method": "signIn",
              "name": "Email and password",
            },
            "className": "css-1r2f04i",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "portis",
        "nodeType": "function",
        "props": Object {
          "auth": Object {
            "Icon": [Function],
            "id": "portis",
            "method": "",
            "name": "Portis wallet",
          },
          "className": "css-1r2f04i",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(h1)>
            Sign Up
             with:
          </Styled(h1)>,
          Array [
            <SignInButton
              auth={
                Object {
                  "Icon": [Function],
                  "id": "google.com",
                  "method": "signInWithGoogle",
                  "name": "Google",
                }
              }
              className="css-1r2f04i"
              onClick={[Function]}
            />,
            <SignInButton
              auth={
                Object {
                  "Icon": [Function],
                  "id": "facebook.com",
                  "method": "signInWithFacebook",
                  "name": "Facebook",
                }
              }
              className="css-1r2f04i"
              onClick={[Function]}
            />,
            <Styled(Link)
              className="css-4rmlxi"
              replace={false}
              to="/main/login/email"
            >
              <SignInButton
                auth={
                  Object {
                    "Icon": [Function],
                    "id": "password",
                    "method": "signIn",
                    "name": "Email and password",
                  }
                }
                className="css-1r2f04i"
              />
            </Styled(Link)>,
            <SignInButton
              auth={
                Object {
                  "Icon": [Function],
                  "id": "portis",
                  "method": "",
                  "name": "Portis wallet",
                }
              }
              className="css-1r2f04i"
              onClick={[Function]}
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Sign Up",
              " with:",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Sign Up",
            " with:",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "google.com",
          "nodeType": "function",
          "props": Object {
            "auth": Object {
              "Icon": [Function],
              "id": "google.com",
              "method": "signInWithGoogle",
              "name": "Google",
            },
            "className": "css-1r2f04i",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "facebook.com",
          "nodeType": "function",
          "props": Object {
            "auth": Object {
              "Icon": [Function],
              "id": "facebook.com",
              "method": "signInWithFacebook",
              "name": "Facebook",
            },
            "className": "css-1r2f04i",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "password",
          "nodeType": "class",
          "props": Object {
            "children": <SignInButton
              auth={
                Object {
                  "Icon": [Function],
                  "id": "password",
                  "method": "signIn",
                  "name": "Email and password",
                }
              }
              className="css-1r2f04i"
            />,
            "className": "css-4rmlxi",
            "replace": false,
            "to": "/main/login/email",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "auth": Object {
                "Icon": [Function],
                "id": "password",
                "method": "signIn",
                "name": "Email and password",
              },
              "className": "css-1r2f04i",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "portis",
          "nodeType": "function",
          "props": Object {
            "auth": Object {
              "Icon": [Function],
              "id": "portis",
              "method": "",
              "name": "Portis wallet",
            },
            "className": "css-1r2f04i",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
